src/models/base/ScholarshipModel.ts

```diff
@@ -0,0 +1,13 @@
+// import firebase from 'firebase/app';
+import FirestoreModel from './FirestoreModel';
+
+export default class ScholarshipModel<
+  T extends { lastModified?: Date; dateAdded?: Date }
+> extends FirestoreModel<T> {
```

Comment:
Why does this class have generics? What does the *T* mean in **ScholarshipModel<T>**? What are the possible values of **T**? If it's only **ScholarshipData** then why not write this like this:
```ts
export default class ScholarshipModel extends FirestoreModel<ScholarshipData> {
```

Reply:
I did this because I was running into a dependency cycle but I was able to fix it by moving the `ScholarshipData` into it's own file located in `types/ScholarshipData.ts`